cep-??.md

```diff
@@ -0,0 +1,246 @@
+
+## Mini language
+
+The `MatchSpec` mini language has gone through several iterations.
+
+The simplest form merely consists of up to three positional arguments: `name [version [build]]`. Only `name` is required. `version` can be any version specifier. `build` can be any string matcher. See "Match conventions" below.
+
+The positional syntax also allows the `=` character as a separator, instead of a space. When this is the case, versions are interpreted differently. `pkg=1.8` will be taken as `1.8.*` (fuzzy), but `pkg 1.8` will give `1.8` (exact). To have fuzzy matches with the space syntax, you need to use `pkg =1.8`. This nuance does not apply if a `build` string is present; both `foo==1.0=*` and `foo=1.0=*` are equivalent (they both understand the version as `1.0`, exact).
+
+`MatchSpec` queries can be also specified with keyword arguments between square brackets. Key-value pairs can be delimited by comma, space, or comma+space. Value can optionally be wrapped in single or double quotes, but must be wrapped if `value` contains a comma, space, or equal sign. Their values can be quoted with single or double quotes. The accepted keys are:
+
+- `channel` (`str`): Name or URL of a channel
+- `subdir` (`str`): Identifier of the subdir (either platform-specific or `noarch`)
+- `version` (`str`): A version specifier
+- `build` (`str`): Build string or a glob match.
+- `build_number` (`int`): Number of the build
+- `md5` (`str`): MD5 hash of the artifact
+- `sha256` (`str`): SHA256 hash of the artifact
+
+These are also accepted but have reduced utility. Their usage is discouraged:
+
+- `url`
+- `track_features`
+- `features`
+- `license`
+- `license_family`
+- `fn`
+
+When both positional and keyword arguments are used, the keyword arguments override the positional information.
+
+## Canonical representation
+
+Since there are several ways of specifying the same information, a canonical representation is welcome. The currently accepted notation is:
+
+```
+(channel(/subdir)::)name(version(build))[key1='value 1',key2=value2]
+```
+
+where `()` indicate optional fields. The rules for constructing a canonical string
+representation are:
+
+1. `name` (i.e. "package name") is required, but its value can be `*`. Its position is always
+   outside the key-value brackets.
+2. If `version` is an exact version, it goes outside the key-value brackets and is prepended
+   by `==`. If `version` is a "fuzzy" value (e.g. `1.11.*`), it goes outside the key-value
+   brackets with the `.*` left off and is prepended by `=`. Otherwise `version` is included
+   inside key-value brackets.
+3. If `version` is an exact version, and `build` is an exact value, `build` goes outside
+   key-value brackets prepended by a `=`. Otherwise, `build` goes inside key-value brackets.
+   `build_string` is an alias for `build`.
+4. The `namespace` position is being held for a future conda feature.
```

Comment:
The `namespace` does not appear in the string above.

Reply:
Thanks, Matt, added!